src/fsharp/TastOps.fs

```diff
@@ -7826,8 +7826,14 @@ let rec mkCompiledTuple g isStruct (argtys,args,m) =
 let mkILMethodSpecForTupleItem (_g : TcGlobals) (typ:ILType) n = 
     mkILNonGenericInstanceMethSpecInTy(typ, (if n < goodTupleFields then "get_Item"+(n+1).ToString() else "get_Rest"), [], mkILTyvarTy (uint16 n))
 
-let mkGetTupleItemN g m n typ te retty =
-    mkAsmExpr([IL.mkNormalCall(mkILMethodSpecForTupleItem g typ n)],[],[te],[retty],m)
+let mkILFieldSpecForTupleItem typ n = 
+    mkILFieldSpecInTy (typ,(if n < goodTupleFields then "Item"+(n+1).ToString() else "Rest"), mkILTyvarTy (uint16 n))
+
+let mkGetTupleItemN g m n (typ:ILType) isStruct te retty =
+    if isStruct then
+        mkAsmExpr([mkNormalLdfld  (mkILFieldSpecForTupleItem typ n)   ],[],[],[retty],m)
```

Comment:
You're not using ``te`` (the expression for the object from which you're loading the field) in the first branch